src/Umbraco.Core/Mapping/UmbracoMapper.cs

```diff
@@ -345,7 +345,8 @@ private TTarget MapEnumerableInternal<TTarget>(IEnumerable source, Type targetGe
 
             if (_ctors.ContainsKey(sourceType))
                 foreach (var c in sourceCtor)
-                    _ctors[sourceType].Add(c.Key, c.Value);
```

Comment:
alternatively, `_ctors[sourceType].TryAdd(c.Key, c.Value)` works and may be cleaner

Reply:
ah of course, didn't see it was a concurrent dictionary, thanks

Reply:
aha, that doesn't actually work since the internal dictionary isn't a concurrent dictionary

Reply:
uh? just pulled latest v8/dev and `_ctors` seems to be a concurrent dictionary? and then,

    if (_ctors.ContainsKey(sourceType))
        foreach (var c in sourceCtor)
            if (!_ctors[sourceType].TryGetValue(c.Key, out _))
                _ctors[sourceType].Add(c.Key, c.Value);
    else
        _ctors[sourceType] = sourceCtor;

is... strange, because that `else` attaches to the second `if`?

maybe I'm not looking at the right place, but that code ... well, R# is not happy

Reply:
_ctors is concurrent dictionary: 

```cs
ConcurrentDictionary<Type, Dictionary<Type, Func<object, MapperContext, object>>>
```

so _ctors[sourceType] is not a concurrent dictionary, it is a normal dictionary

Reply:
and yes stupid brackets 😱 will update

Reply:
aha - all good - and better to update since... with the `else` thing, there's a bug now

Reply:
If `_ctors` is a `ConcurrentDictionary`, why doesn't it use `TryAdd` instead of the `ContainsKey`/set _and_ is updating values of the internal dictionary using `_ctors[sourceType]` (e.g. not thread-safe)?